lms/djangoapps/discussion_api/forms.py

```diff
@@ -2,12 +2,38 @@
 Discussion API forms
 """
 from django.core.exceptions import ValidationError
-from django.forms import BooleanField, CharField, Form, IntegerField, NullBooleanField
+from django.forms import (
+    BooleanField,
+    CharField,
+    Field,
+    Form,
+    IntegerField,
+    MultipleHiddenInput,
+    NullBooleanField,
+)
 
 from opaque_keys import InvalidKeyError
 from opaque_keys.edx.locator import CourseLocator
 
 
+class TopicIdField(Field):
+    """
+    Field for a list of topic_ids
+    """
+    def __init__(self, *args, **kwargs):
+        self.field_type = CharField()
+        self.widget = MultipleHiddenInput
+        super(TopicIdField, self).__init__(*args, **kwargs)
+
+    def clean(self, value):
+        if value:
+            for subvalue in value:
+                self.field_type.validate(subvalue)
+            return [self.field_type.clean(subvalue) for subvalue in value]
```

Comment:
What's the goal of calling clean on each subvalue? It looks to me like the only thing it will do is raise a misleading ValidationError if one of the values is empty.